### ISSUE TITLE
Issues when changing connections

### ISSUE THREAD

**Issue by kdrtoona** [microsoft/azuredatastudio#15616]
Issue Type: <b>Bug</b>

When changing server connections sometimes the connection just spins and either never connects or it connects and you can not change to another database.  When this happens if you try running a script ADS will lock up and you have to quit and re-open the application. 

Azure Data Studio version: azuredatastudio 1.29.0 (0f5cfdc2c8c2433028842e16dc5ac1a65da24292, 2021-05-15T01:12:11.112Z)
OS version: Windows_NT x64 10.0.19041

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10610U CPU @ 1.80GHz (8 x 2304)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.61GB (6.98GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (13)</summary>

Extension|Author (truncated)|Version
---|---|---
sqldm-performance-insights|IDE|0.4.0
admin-pack|Mic|0.0.2
admin-tool-ext-win|Mic|0.0.2
agent|Mic|0.48.0
cms|Mic|0.8.0
dacpac|Mic|1.8.0
datavirtualization|Mic|1.12.0
import|Mic|1.3.0
powershell|ms-|2020.4.0
profiler|Mic|0.11.0
query-history|Mic|0.1.0
sql-prompt|Red|0.2.7
sql-search|Red|0.4.0

</details>

**Comment by abist** [comment]
@kdrtoona Thanks for opening the issue. Could you give more details on what type of connection it is? Do you see any errors in the UI when the connection fails?

**Comment by kdrtoona** [comment]
@abist  Certainly.  The connections are SQL Server connections.  Very often I need to move from one environment to another to test my SQL, when doing so I click on Change Connection which brings up the Connection frame.  I have several saved connections to various SQL servers.  I click on the connection I want and it usually appears to connect. However, on occasion, after changing the connection, I will go to the Database Selection drop down and if it lists the databases on the server, I'm probably good to go; if it lists only one database (the default) and I run a script, the script will just appear to run and run and run and I eventually have to quit ADS and re-open.  I rarely if ever get an error message, and if I do it says Unable to Change Database or something like that but on those occasions, I'm usually able to run my script and it will work without issue.  
My current workaround for this is to just keep changing connections back and forth until it brings up the full database list in the drop down in the environment I need to use.

**Comment by abist** [comment]
@kdrtoona Added some instructions to retrieve logs for us to investigate further

**Comment by kdrtoona** [comment]
@abist  Here you go.  I was able to get it to generate an error message this time. 

[20210602T170031.zip](https://github.com/microsoft/azuredatastudio/files/6600853/20210602T170031.zip)
[console.log](https://github.com/microsoft/azuredatastudio/files/6600857/console.log)

**Comment by abist** [comment]
@kdrtoona Thanks, this was helpful!
